api/src/test/java/org/apache/iceberg/types/TestTypes.java

```diff
@@ -81,6 +81,56 @@ public void fromPrimitiveString() {
         .withMessage("Cannot parse type string to primitive: abcdefghij");
   }
 
+  @Test
+  public void geospatialTypeFromTypeName() {
+    assertThat(Types.fromPrimitiveString("geometry")).isEqualTo(Types.GeometryType.crs84());
+    assertThat(Types.fromPrimitiveString("geometry()")).isEqualTo(Types.GeometryType.crs84());
```

Comment:
Why is this a valid? It seems like a problem to me. We don't allow `long()`.

Reply:
Fixed. `geometry()` and `geography()` are not valid representations of geospatial types.